src/utils/actions.ts

```diff
@@ -0,0 +1,13 @@
+export type ActionWithPayload<T, P> = Action<T> & {
+  payload: P;
+};
+
+export type Action<Type> = {
```

Comment:
I'd recommend this being at the top, as it's used by `ActionWithPayload`.
Typescript may not exist, but still treat it like it does and try to ensure type declaration ordering is correct.